Server/src/RedisManager.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024 Microsoft, Inc. and others.  All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ */
+
+#include "RedisManager.h"
+
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::RedisManager() {
+    exit_ = false;
+}
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::~RedisManager() {
+}
+
+
+/*********************************************************************
+ * Setup for this class
+ *
+ * \param [in] logPtr     logger pointer
+ * \param [in] cfgPtr     config pointer
+ ***********************************************************************/
+void RedisManager::Setup(Logger *logPtr, Config *cfgPtr) {
+    logger = logPtr;
+    if (!cfgPtr->redis_multiAsic) {
+        if (!swss::SonicDBConfig::isInit()) {
+            swss::SonicDBConfig::initialize();
+        }
+    } else {
+        if (!swss::SonicDBConfig::isGlobalInit()) {
+            swss::SonicDBConfig::initializeGlobalConfig();
+        }
+    }
+    stateDb_ =  std::make_shared<swss::DBConnector>(BMP_DB_NAME, 0, false);
+    separator_ = swss::SonicDBConfig::getSeparator(BMP_DB_NAME);
+}
+
+
+
+/**
+ * Get Key separator for deletion
+ *
+ * \param [in] N/A
+ */
+std::string RedisManager::GetKeySeparator() {
+    return separator_;
+}
+
+
+/**
+ * WriteBMPTable
+ *
+ * \param [in] table            Reference to table name
+ * \param [in] key              Reference to various keys list
+ * \param [in] fieldValues      Reference to field-value pairs
+ */
+bool RedisManager::WriteBMPTable(const std::string& table, const std::vector<std::string>& keys, const std::vector<swss::FieldValueTuple> fieldValues) {
+
+    if (enabledTables_.find(table) == enabledTables_.end()) {
+        LOG_INFO("RedisManager %s is disabled", table.c_str());
+        return false;
+    }
+    std::unique_ptr<swss::Table> stateBMPTable = std::unique_ptr<swss::Table>(new swss::Table(stateDb_.get(), table));
+    std::string fullKey;
+    for (const auto& key : keys) {
+        fullKey += key;
+        fullKey += separator_;
+    }
+    fullKey.erase(fullKey.size() - 1);
+
+    DEBUG("RedisManager WriteBMPTable key = %s", fullKey.c_str());
+
+    stateBMPTable->set(fullKey, fieldValues);
+    return true;
+}
+
+
+/**
+ * RemoveEntityFromBMPTable
+ *
+ * \param [in] keys             Reference to various keys
+ */
+bool RedisManager::RemoveEntityFromBMPTable(const std::vector<std::string>& keys) {
+
+    stateDb_->del(keys);
```

Comment:
>stateDb_->del(keys) [](http://example.com/codeflow?start=4&length=19)

Possible add error handling and corresponding log when we do removal.

Reply:
seems swss-common has no error as well, just triggers RedisCommand https://github.com/liuh-80/sonic-swss-common/blob/898aa5dbee22920847dafb0849907d55c8a84816/common/dbconnector.cpp#L1012

Reply:
I understand it is in redis command mode, but it can be crashed during any stack of call chain, to keep this RedisManager keeps to running, I'd suggest we do error handling.

Reply:
refer https://github.com/sonic-net/sonic-swss-common/blob/bd0f2eb0613aba8a448645b3bf53a44a50bd763f/common/dbconnector.cpp#L1019, this swss-common api is void.
![image](https://github.com/user-attachments/assets/e914cb97-895a-46f0-869b-81cda1d87abc)